### PR TITLE
Feature/client/commands improvements

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
Create all command for improvment this code is going to be change with refacto of commands

### CODE REVIEW

File: client/src/get_commands/get_cmd_logout.c [hunk 1]
```diff
@@ -6,31 +6,34 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
         printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_logout.c:36]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_logout.c [hunk 2]
```diff
@@ -6,31 +6,34 @@
...
-        client_reply(atoi(buff));
-        // atoi -> compare error code
-    } else  {
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_logout.c:32]: What if the call system faills ?

File: client/src/get_commands/get_cmd_send_message.c [hunk 3]
```diff
@@ -6,26 +6,34 @@
...
-        // atoi -> compare error code
-    } else  {
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_send_message.c:33]: What if the call system faills ?

File: client/src/get_commands/get_cmd_send_message.c [hunk 4]
```diff
@@ -6,26 +6,34 @@
...
+        }
+    } else {
         printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_send_message.c:36]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_subscribe_teams.c [hunk 5]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_subscribe_teams.c:36]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_subscribe_teams.c [hunk 6]
```diff
@@ -5,3 +5,35 @@
...
+        make_command_rfc_compatible(buff, "SUB ", av);
+        write(socket, buff, strlen(buff));
+        memset(buff, 0, 4096);
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_subscribe_teams.c:32]: What if the call system faills ?

File: client/src/get_commands/get_cmd_unsubscribe_teams.c [hunk 7]
```diff
@@ -5,3 +5,35 @@
...
+        make_command_rfc_compatible(buff, "USUB ", av);
+        write(socket, buff, strlen(buff));
+        memset(buff, 0, 4096);
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_unsubscribe_teams.c:32]: What if the call system faills ?

File: client/src/get_commands/get_cmd_unsubscribe_teams.c [hunk 8]
```diff
@@ -5,3 +5,35 @@
...
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_unsubscribe_teams.c:36]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_create_commands/get_cmd_create_channel.c [hunk 9]
```diff
@@ -7,3 +7,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_channel.c:38]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_create_commands/get_cmd_create_channel.c [hunk 10]
```diff
@@ -7,3 +7,35 @@
...
+        make_command_rfc_compatible(buff, "CRCH ", av);
+        write(socket, buff, strlen(buff));
+        memset(buff, 0, 4096);
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_channel.c:34]: What if the call system faills ?

File: client/src/get_commands/get_create_commands/get_cmd_create_teams.c [hunk 11]
```diff
@@ -7,3 +7,35 @@
...
+        make_command_rfc_compatible(buff, "CRTE ", av);
+        write(socket, buff, strlen(buff));
+        memset(buff, 0, 4096);
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_teams.c:34]: What if the call system faills ?

File: client/src/get_commands/get_create_commands/get_cmd_create_teams.c [hunk 12]
```diff
@@ -7,3 +7,35 @@
...
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_teams.c:38]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_create_commands/get_cmd_create_thread.c [hunk 13]
```diff
@@ -7,3 +7,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_thread.c:38]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_create_commands/get_cmd_create_thread.c [hunk 14]
```diff
@@ -7,3 +7,35 @@
...
+        make_command_rfc_compatible(buff, "CRTH ", av);
+        write(socket, buff, strlen(buff));
+        memset(buff, 0, 4096);
+        if (read(socket, buff, 4096) == 0) {
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
```
Comment (ZQUEMA) [inline client/src/get_commands/get_create_commands/get_cmd_create_thread.c:34]: What if the call system faills ?

File: client/src/get_commands/get_info_commands/get_cmd_get_user_infos.c [hunk 15]
```diff
@@ -6,3 +6,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_info_commands/get_cmd_get_user_infos.c:40]: Same as above

File: client/src/get_commands/get_info_commands/get_cmd_infos_channel.c [hunk 16]
```diff
@@ -4,3 +4,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_info_commands/get_cmd_infos_channel.c:38]: Same as above

File: client/src/get_commands/get_info_commands/get_cmd_infos_team.c [hunk 17]
```diff
@@ -5,3 +5,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_info_commands/get_cmd_infos_team.c:39]: Same as above

File: client/src/get_commands/get_info_commands/get_cmd_infos_thread.c [hunk 18]
```diff
@@ -5,3 +5,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_info_commands/get_cmd_infos_thread.c:39]: Same as above

File: client/src/get_commands/get_list_commands/get_cmd_list_channels.c [hunk 19]
```diff
@@ -5,3 +5,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_list_commands/get_cmd_list_channels.c:39]: Same as above

File: client/src/get_commands/get_list_commands/get_cmd_list_threads.c [hunk 20]
```diff
@@ -5,3 +5,35 @@
...
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_list_commands/get_cmd_list_threads.c:39]: Same as above

File: client/src/get_commands/get_list_commands/get_cmd_list_user_subscribed.c [hunk 21]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_list_commands/get_cmd_list_user_subscribed.c:36]: Same as above

File: client/src/get_commands/get_list_commands/get_cmd_list_users.c [hunk 22]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_list_commands/get_cmd_list_users.c:36]: Same as above

File: client/src/get_commands/get_list_commands/get_cmd_list_users_messages.c [hunk 23]
```diff
@@ -5,3 +5,35 @@
...
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
+    }
```
Comment (ZQUEMA) [inline client/src/get_commands/get_list_commands/get_cmd_list_users_messages.c:37]: Same as above

File: client/src/get_commands/get_use_commands/get_cmd_use_channel.c [hunk 24]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_use_commands/get_cmd_use_channel.c:36]: Same as above

File: client/src/get_commands/get_use_commands/get_cmd_use_teams.c [hunk 25]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_use_commands/get_cmd_use_teams.c:36]: Same as above

File: client/src/get_commands/get_use_commands/get_cmd_use_threads.c [hunk 26]
```diff
@@ -5,3 +5,35 @@
...
+            printf("Client is deconnected !\n");
+            exit (0);
+        }
+        client_reply(atoi(strncpy(response, buff, 2)));
+        if (strlen(buff) > 4) {
+            tab_response = parse_response(buff, 2);
+            client_event_logged_in(tab_response[1], tab_response[3]);
+        }
+    } else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_use_commands/get_cmd_use_threads.c:36]: Same as above

File: rfc11720 [hunk 27]
```diff
@@ -104,6 +153,40 @@ The objectives of the myTeams project RFC is to allow communication between myTe
     - Invalid format
 - 24
     - Resource already exists
+- 25
+    - Client does not log/exist in database
```
Comment (aurelenc) [inline rfc11720:157]: 14: Not found

File: rfc11720 [hunk 28]
```diff
@@ -92,6 +139,8 @@ The objectives of the myTeams project RFC is to allow communication between myTe
     - Forbidden
 - 14
     - Not found
+- 15
+    - Logout or Lost Connexion
```
Comment (aurelenc) [inline rfc11720:143]: The server can't send this code if the client lost connection

File: rfc11720 [hunk 29]
```diff
@@ -81,6 +124,10 @@ The objectives of the myTeams project RFC is to allow communication between myTe
     - Success
 - 01
     - Client connected
+- 02
+    - Client subscribe at a teams OK
```
Comment (aurelenc) [inline rfc11720:128]: We should only have one "Success" code

File: rfc11720 [hunk 30]
```diff
@@ -104,6 +153,40 @@ The objectives of the myTeams project RFC is to allow communication between myTe
     - Invalid format
 - 24
     - Resource already exists
+- 25
+    - Client does not log/exist in database
+
+
+- 30
+    - Unknown team
```
Comment (aurelenc) [inline rfc11720:161]: 14: Not found ?

File: rfc11720 [hunk 31]
```diff
@@ -104,6 +153,40 @@ The objectives of the myTeams project RFC is to allow communication between myTe
...
+    - Unknown channel
+- 32
+    - Unknown thread
+- 33
+    - Unknown user
+
+
+
+- 40
+    - Print all users ok
```
Comment (aurelenc) [inline rfc11720:172]: 00: Success ?

File: rfc11720 (hunk 27 above)
Comment (TomDUVAL-MAHE) [reply rfc11720:157]: ok

File: rfc11720 (hunk 30 above)
Comment (TomDUVAL-MAHE) [reply rfc11720:161]: Nop i wan't to know what is not found :D

File: rfc11720 (hunk 31 above)
Comment (TomDUVAL-MAHE) [reply rfc11720:172]: Nop i wan't to know what is succes

File: client/Makefile [hunk 32]
```diff
@@ -7,21 +7,34 @@
...
-		src/get_commands/get_cmd_login.c	\
-		src/get_commands/get_cmd_logout.c	\
 		src/get_commands/check_params.c	\
 		src/get_commands/manage_params.c	\
-		src/get_commands/get_cmd_send_message.c	\
 		src/manage_commands/make_command_rfc_compatible.c	\
 		src/manage_commands/parse_server_response.c \
-		src/manage_commands/reply_codes.c
+		src/manage_commands/reply_codes.c	\
+		src/manage_commands/send_command.c
```
Comment (ZQUEMA) [inline client/Makefile:33]: What is the difference between them and the ones above? They both handle commands

File: client/src/get_commands/get_cmd_create.c [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+    // char const *reply_body);
+
+
+
+
+
+
+    free(tab_response);
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_create.c:105]: The function has to be conform to Epitech C coding style

File: client/include/get_command.h [hunk 34]
```diff
@@ -13,9 +13,21 @@
...
 int send_message(char *av, int socket);
+int logout_client(char *av, int socket);
+int login_client(char *av, int socket);
+int use(char *av, int socket);
+int info(char *av, int socket);
+int list(char *av, int socket);
+int create(char *av, int socket);
+int messages(char *av, int socket);
+int user(char *av, int socket);
+int users(char *av, int socket);
```
Comment (ZQUEMA) [inline client/include/get_command.h:30]: Documentation is needed to understand how they work and what the parameters are (av ?).

File: client/include/get_command.h [hunk 35]
```diff
@@ -30,6 +42,8 @@ int check_params(char *av);
 
 char **parse_response(char *response, int nb_params);
 
+char **send_command(char *av, char **tab_response, char *command, int socket);
+
 int parse_command(char *input, int socket);
```
Comment (ZQUEMA) [inline client/include/get_command.h:47]: Same here

File: client/src/get_commands/get_cmd_messages.c [hunk 36]
```diff
@@ -0,0 +1,41 @@
...
+    // time_t message_timestamp,
+    // char const *message_body);
+
+    if (!strcmp(code_response, "??"))
+        client_error_unknown_user(tab_response[0]);
+    // char const *user_uuid);
+
+    free(tab_response);
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_messages.c:41]: The function has to be conform to Epitech C coding style

File: client/src/get_commands/get_cmd_unsubscribe_teams.c [hunk 37]
```diff
@@ -5,3 +5,33 @@
...
+
+    if (!strcmp(code_response, "03"))
+        client_print_unsubscribed(tab_response[1], tab_response[3]);
+
+    if (!strcmp(code_response, "30"))
+        client_error_unknown_team(tab_response[1]);
+
+
+    free(tab_response);
+    return 0;
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_unsubscribe_teams.c:36]: The function has to be conform to Epitech C coding style

File: client/src/get_commands/get_cmd_info.c [hunk 38]
```diff
@@ -0,0 +1,70 @@
...
+
+int info(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+    if (check_params(av) == 0)
+        tab_response = send_command(av, tab_response, "INFO ", socket);
+    else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_info.c:21]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_login.c [hunk 39]
```diff
@@ -12,28 +12,20 @@
...
-        if (strlen(buff) > 4) {
-            tab_response = parse_response(buff, 2);
-            client_event_logged_in(tab_response[1], tab_response[3]);
-        }
-    } else {
+
+    if (check_params(av) == 1)
+        tab_response = send_command(av, tab_response, "LOGI ", socket);
+    else {
         printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_login.c:22]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_messages.c [hunk 40]
```diff
@@ -0,0 +1,41 @@
...
+
+int messages(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+    if (check_params(av) == 1)
+        tab_response = send_command(av, tab_response, "MSG ", socket);
+    else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_messages.c:21]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_use.c [hunk 41]
```diff
@@ -0,0 +1,32 @@
...
+int use(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+
+    if (check_params(av) <= 3)
+        tab_response = send_command(av, tab_response, "USE ", socket);
+    else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_use.c:22]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_use.c [hunk 42]
```diff
@@ -0,0 +1,32 @@
...
+#include "reply_codes.h"
+#include "logging_client.h"
+
+int use(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+
+    if (check_params(av) <= 3)
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_use.c:19]: The function has to be conform to Epitech C coding style

File: client/src/get_commands/get_cmd_user.c [hunk 43]
```diff
@@ -0,0 +1,42 @@
...
+    // int user_status);
+
+    if (!strcmp(code_response, "33"))
+        client_error_unknown_user(tab_response[1]);
+    // char const *user_uuid);
+
+
+    free(tab_response);
+    return 0;
+}
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_user.c:42]: The function has to be conform to Epitech C coding style

File: client/src/get_commands/get_cmd_user.c [hunk 44]
```diff
@@ -0,0 +1,42 @@
...
+
+int user(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+    if (check_params(av) == 1)
+        tab_response = send_command(av, tab_response, "USR ", socket);
+    else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_user.c:21]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_users.c [hunk 45]
```diff
@@ -0,0 +1,37 @@
...
+
+int users(char *av, int socket)
+{
+    char code_response[3];
+    char **tab_response = NULL;
+
+    if (check_params(av) == 0)
+        tab_response = send_command(av, tab_response, "USRS ", socket);
+    else {
+        printf("Command are not good use /help for more information !\n");
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_users.c:21]: "The given command is not correct. Use /help for more information !\n"

File: client/src/get_commands/get_cmd_users.c [hunk 46]
```diff
@@ -0,0 +1,37 @@
...
+    code_response[2] = '\0';
+    if (!strcmp(code_response, "14"))
+        client_error_unauthorized();
+
+    if (!strcmp(code_response, "40")) // @timestamp
+        client_print_users(tab_response[1], tab_response[3], atoi(tab_response[5]));
+    // char const *user_uuid,
+    // char const *user_name,
+    // int user_status);
+
```
Comment (ZQUEMA) [inline client/src/get_commands/get_cmd_users.c:34]: The function has to be conform to Epitech C coding style

File: rfc11720 [hunk 47]
```diff
@@ -9,70 +9,113 @@ The objectives of the myTeams project RFC is to allow communication between myTe
...
+- INFO <EOL>
     -   This protocol is used to display currently selected channel infos
-- IFTH <EOL>
+
+- INFO <EOL>
     -   This protocol is used to display currently selected thread infos
-- IFUS <EOL>
+
+- INFO <EOL>
     -   This protocol is used to display currently logged user infos
```
Comment (ZQUEMA) [inline rfc11720:119]: It needs to be resumed. There is only one protocol but the uses will be different with the USE context

File: rfc11720 (hunk 29 above)
Comment (ZQUEMA) [reply rfc11720:128]: We must have only one code "Success". Maybe the type should be given in the parameters of the answer.

File: rfc11720 (hunk 28 above)
Comment (ZQUEMA) [reply rfc11720:143]: It should be up to the client to detect if the connection to the server is broken

**Review by ZQUEMA — CHANGES_REQUESTED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**